templates/repo/home.tmpl

```diff
@@ -29,26 +29,25 @@
 			{{end}}
 		</div>
 		<div class="gt-mt-3" id="repo-topics">
-		{{range .Topics}}<a class="ui repo-topic large label topic" href="{{AppSubUrl}}/explore/repos?q={{.Name}}&topic=1">{{.Name}}</a>{{end}}
-		{{if and .Permission.IsAdmin (not .Repository.IsArchived)}}<a id="manage_topic" class="muted">{{.locale.Tr "repo.topic.manage_topics"}}</a>{{end}}
+			{{range .Topics}}<a class="ui repo-topic large label topic" href="{{AppSubUrl}}/explore/repos?q={{.Name}}&topic=1">{{.Name}}</a>{{end}}
+			{{if and .Permission.IsAdmin (not .Repository.IsArchived)}}<button id="manage_topic" class="ui button tiny tertiary">{{.locale.Tr "repo.topic.manage_topics"}}</button>{{end}}
 		</div>
 		{{end}}
 		{{if and .Permission.IsAdmin (not .Repository.IsArchived)}}
-		<div class="ui repo-topic-edit grid form gt-hidden" id="topic_edit">
-			<div class="fourteen wide column">
-				<div class="field">
-					<div class="ui fluid multiple search selection dropdown">
-						<input type="hidden" name="topics" value="{{range $i, $v := .Topics}}{{.Name}}{{if lt (Add $i 1) (len $.Topics)}},{{end}}{{end}}">
-						{{range .Topics}}
-						<div class="ui small label topic transition visible" data-value="{{.Name}}" style="display: inline-block !important; cursor: default;">{{.Name}}{{svg "octicon-x" 16 "delete icon gt-ml-3 gt-mt-1"}}</div>
-						{{end}}
-						<div class="text"></div>
-					</div>
+		<div class="ui form gt-hidden gt-df gt-mt-4" id="topic_edit">
+			<div class="field gt-f1 gt-mr-3">
+				<div class="ui fluid multiple search selection dropdown">
+					<input type="hidden" name="topics" value="{{range $i, $v := .Topics}}{{.Name}}{{if lt (Add $i 1) (len $.Topics)}},{{end}}{{end}}">
+					{{range .Topics}}
+						{{/* lalels generated by Fomantic still use `<i class="delete icon">`, to avoid mixing different style together and to remind future developers, use it instead of svg*/}}
+						<div class="ui small label topic transition visible" data-value="{{.Name}}" style="display: inline-block !important; cursor: default;">{{.Name}}<i class="delete icon"></i></div>
```

Comment:
Without considering screen reader, current code just works well with keyboard. You can use Left&Right Arrow keys to navigate between labels, then the selected label would be highlighted, and then press Del or Bksp to delete it. So these labels are already somewhat "focusable". 

When considering screen reader .... there are far more problems than the "accessible label".

But I can't test all kinds of screen readers. If you mean that: your screen reader (what's the name?) requires that accessible label on each delete icon, I could try to figure it out later.

Reply:
I see your points. I will do some tests with mobile to see what we could do. I think we can have a clear solution.

Reply:
After doing some quick tests, I think the results are:

> What I'm not entirely sure about is whether Del or Bksp will work on mobile 

On mobile, without screen reader, all operations work.

> I mainly use Talkback  ... the aria role is definitely not a menu but a combobox 

Yup, this is not the only confusing dropdown, there are also the License/Template selection dropdowns on the Create Repo page, etc. I didn't make some of them work well before, and I had to leave a comment in `aria.js`: `TODO: multiple selection is not supported yet`

>  the delete icon is confusing in terms of expectations ... remove it or not?

If I understand correctly, the Fomantic Dropdown's behavior was designed as such, and it looks intuitive to me:

* If a label is selected by keyboard, it's just highlighted (looks like selected text), then the user use Del to delete it.
* If a label itself is selected by mouse clicking, the same as above.
* If a label's `delete icon` is clicked, then it will be deleted immediately.

So I guess adding `aria-hidden` to the `delete icon` could be fine at the moment?

----

How do you think about:

1. Add the `aria-hidden` to these `delete icon` elements
2. Keep current approach (`aria.js` adds `menu` related attributes), or do not touch this dropdown's aria attributes (just make `aria.js` ignore it)
3. Get this PR merged and try to figure out other improvements in the future.

?

Reply:
1. Good, is what I suggested.
2. I am working on suggestion of aria.js differentiation of roles in #22631.
3. Looks good to me without really tested yet. I will remove my fix of this problem from #22631 to avoid conflicts with yours.
 
Just one doubt remains, how do you delete selected topics in mobile with a screen reader?

Reply:
One more thing, I think I misunderstood one point, you might mean:

4. Make the `delete icon` a11y friendly instead of `aria-hidden`, then the old selection/deletion behavior could be kept, and users could click the delete icon to delete the topic label, is it also acceptable?

Reply:
> One more thing, I think I misunderstood one point, you might mean:
> 
>     4. Make the `delete icon` a11y friendly instead of `aria-hidden`, then the old selection/deletion behavior could be kept, and users could click the `delete icon` to delete the topic label (replace my point 1), is it also acceptable?

yes, I think it would be much better, and probably will give a solution to sr over mobile

Reply:
Could you have a try with latest commit? (I think now I might be able to understand your requirements about 80-90%, before the discussion, I never thought it's related to Mobile and the Delete Icon 😂 ).

Feel free to describe the problems with more details.

Now the existing labels and newly-added labels all have aria-related attributes.

![image](https://user-images.githubusercontent.com/2114189/223465031-3f16bdb1-5f84-48ce-b790-16ec7d261680.png)

Reply:
Ok, but please bear with me as I'm in the middle of another task.

In a quick view, is there any way for existing and new labels to have the same markup? I suspect the new tags will be left with two focus points on the anchor and the button, which is annoying. If I remember correctly there is an ajax request "complete" handler (or something similar) that defines the markup of the added elements. Perhaps you could review that code to get a better view on it.

Reply:
No hurry, take your time.

>  is there any way for existing and new labels to have the same markup? 

Yes, see my screenshot, old and new labels all get correct aria-related attributes now.

I have tested it with my Android (see the second screenshot), I think it works.

The key point is the calls to `addLabelDeleteIconAria`, especially the one in `onLabelCreate`.

Reply:
> Yes, see my screenshot, existing and new labels all get correct aria-related attributes now, and they have the same markup now.

I was talking about the labels; in screenshot 1, the existing label is a `div` and the new one is an `a` (anchor). You are talking about '<i>' apparently.

Reply:
It has been fixed by latest commit, I haven't updated the screenshot. Will update the screenshot